packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -182,6 +185,13 @@ function reducer<T, C>(data: AsyncListState<T, C>, action: Action<T, C>): AsyncL
             items: action.type === 'loading' ? [] : data.items,
             abortController: action.abortController
           };
+        case 'update':
+          // We're already loading, and an update happened at the same time (e.g. selectedKey changed).
+          // Update data but don't abort previous load.
+          return {
+            ...data,
+            ...action.updater(data)
+          };
```

Comment:
Don't abort previous load/loadMore/filter otherwise we won't get the filtered list/new items.
Example Scenario:
1. User types in the field triggering a `filtering` action. `dispatch({...action, abortController})` happens, updating the data state with the new filter text. fetch request hasn't finished
2. `list.setSelectedKey` is called, triggering a `update` action. 
3.  if we call `data.abortController.abort();` here we would end up with an up to date `data.filterText` and `data.selectedKey`, but the fetch from step 1 will be aborted and our list of items will be out of date